src/main/java/com/zerobase/mentalgrowhdiary/security/TokenProvider.java

```diff
@@ -0,0 +1,130 @@
+package com.zerobase.mentalgrowhdiary.security;
+
+
+import com.zerobase.mentalgrowhdiary.service.UserService;
+import com.zerobase.mentalgrowhdiary.type.Role;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.SecretKey;
+import java.util.Base64;
+import java.util.Date;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class TokenProvider {
+
+    private static final long TOKEN_EXPIRE_TIME = 1000 * 60 * 60 * 24;
+    private static final String KEY_ROLE = "role";
+
+    @Value("${spring.jwt.secret}")
+    private String secretKeyEncoded;
+
+    private final UserService userService;
+    private SecretKey secretKey;
+
+
+
+    @PostConstruct
+    public void init(){
+
+        try {
+            byte[] decodedKey = Base64.getDecoder().decode(secretKeyEncoded);
+            this.secretKey = Keys.hmacShaKeyFor(decodedKey);
+            log.info("TokenProvider initialized success.");
+
+        } catch (IllegalArgumentException e) {
+
+            log.error("Failed to initialize TokenProvider : ", secretKeyEncoded);
```

Comment:
```suggestion
            log.error("Failed to initialize TokenProvider : {}", secretKeyEncoded);
```
{}가 있어야 , 뒤의 데이터가 {}에 치환됩니다